sigmastate/src/test/scala/sigmastate/PrettyPrintErgoTreeSpecification.scala

```diff
@@ -0,0 +1,554 @@
+package sigmastate
+
+import org.ergoplatform.MinerPubkey
+import org.ergoplatform.settings.ErgoAlgos
+import sigmastate.eval.IRContext
+import sigmastate.interpreter.Interpreter
+import sigmastate.serialization.ErgoTreeSerializer.DefaultSerializer
+import sigmastate.utxo.SelectField
+import sigmastate.Values.{BigIntConstant, BlockValue, ByteArrayConstant, GroupGenerator, SValue, Tuple, ValDef, ValUse}
+import special.sigma.SigmaDslTesting
+
+class PrettyPrintErgoTreeSpecification extends SigmaDslTesting {
+  implicit def IR: IRContext = createIR()
+  private def compile(code: String, env: Interpreter.ScriptEnv = Interpreter.emptyEnv): SValue = compile(env, code)
+
+  property("block value"){
+    val node = BlockValue(
+      Array(
+        ValDef(1, List(), Tuple(1, 5)),
+        ValDef(3, List(), SelectField(ValUse(1, STuple(SInt, SInt)), 1)),
+        ValDef(4, List(), SelectField(ValUse(1, STuple(SInt, SInt)), 2))
+      ),
+      Plus(ValUse(3, SInt), ValUse(4, SInt))
+    )
+    PrettyPrintErgoTree.prettyPrint(node) shouldBe
+      """{
+        |  val tuple1 = (1, 5)
+        |  val i3 = tuple1._1
+        |  val i4 = tuple1._2
+        |  i3 + i4
+        |}""".stripMargin
+  }
+
+  property("booleans"){
+    val code = "{ (x: (Int, Boolean)) => ((x._1 == 0) ^ x._2) && (false || true) }"
+    val compiledTree = compile(code)
+    PrettyPrintErgoTree.prettyPrint(compiledTree, 60) shouldBe
+      """{{(tuple1: (Int, Boolean)) =>
+        |  ((tuple1._1 == 0) ^ tuple1._2) && (false || true)
+        |}}""".stripMargin
+  }
+
+  property("reading context of register as typed value"){
+    val code = "{SELF.R4[Coll[Box]].get}"
+    val compiledTree = compile(code)
+    PrettyPrintErgoTree.prettyPrint(compiledTree) shouldBe code
+  }
+
+  property("function with block values in body"){
+    val code =
+      """{ (x: (Short, Short)) =>
+        |  val a = x._1
+        |  val b = x._2
+        |  val plus = a + b
+        |  val minus = a - b
+        |  val mul = a * b
+        |  val div = a / b
+        |  val mod = a % b
+        |  val minimum = min(a, b)
+        |  val maximum = max(a, b)
+        |  (plus, (minus, (mul, (div, (mod, (minimum, maximum))))))
+        |}""".stripMargin
+    val compiledTree = compile(code)
+    PrettyPrintErgoTree.prettyPrint(compiledTree, 100) shouldBe
+      """{{(tuple1: (Short, Short)) =>
+        |  val s3 = tuple1._1
+        |  val s4 = tuple1._2
+        |  (s3 + s4, (s3 - s4, (s3 * s4, (s3 / s4, (s3 % s4, (min(s3, s4), max(s3, s4)))))))
+        |}}""".stripMargin
+    
+  }
+
+  property("branching"){
```

Comment:
I think we need to handle cases where if statements are inlined in binops, here's a test case:

```
  property("inline-if"){
    val code =
      """{
        | val a = if (SELF.value > 1000) { 1000 } else { 1500 }
        |
        | 50 > a
        |}""".stripMargin
    val compiledTree = compile(code)
    PrettyPrintErgoTree.prettyPrint(compiledTree, 60) shouldBe
      """{
        | val a = if (SELF.value > 1000) { 1000 } else { 1500 }
        |
        | 50 > a
        |}""".stripMargin
  }
```

Currently `prettyPrint` produces which doesn't seem valid:

```
{[50 > if (SELF.value > 1000L) { 1000 } else { 1500 ]}}
```